addOns/graphql/src/main/java/org/zaproxy/addon/graphql/GraphQlParam.java

```diff
@@ -170,6 +171,17 @@ public ArgsTypeOption getArgsType() {
         return argsType;
     }
 
+    // For generating an API action.
+    public void setArgsType(String argsType) {
+        try {
+            setArgsType(ArgsTypeOption.valueOf(argsType.toUpperCase(Locale.ROOT)));
+        } catch (IllegalArgumentException e) {
+            LOG.error('"' + argsType + "\" is not a valid Arguments Specification Type.");
+            // Rethrow for API.
+            throw e;
```

Comment:
It should throw as `ApiException`, otherwise this will be handled as an internal error.